crates/interledger-settlement-engines/src/engines/ethereum_ledger/eth_engine.rs

```diff
@@ -504,11 +510,64 @@ where
-                    if response.status().is_success() {
-                        Ok(())
-                    } else {
-                        Err(())
+                    if !response.status().is_success() {
+                        return Either::A(err(()));
                     }
+                    Either::B(
+                        response
+                            .into_body()
+                            .concat2()
+                            .map_err(|err| {
+                                let err = format!("Couldn't retrieve body {:?}", err);
+                                error!("{}", err);
+                            })
+                            .and_then(move |body| {
+                                // Get the amount stored by the connector and
+                                // check for any precision loss / overflow
+                                serde_json::from_slice::<Quantity>(&body).map_err(|err| {
+                                    let err = format!(
+                                        "Couldn't parse body {:?} into Quantity {:?}",
+                                        body, err
+                                    );
+                                    error!("{}", err);
+                                })
+                            })
+                            .and_then(move |quantity: Quantity| {
+                                // Convert both to BigUInts so we can do arithmetic
+                                join_all(vec![
+                                    result(BigUint::from_str(&quantity.amount)),
+                                    result(BigUint::from_str(&amount.to_string())),
+                                ])
+                                .map_err(|err| {
+                                    let error_msg =
+                                        format!("Error converting to BigUints {:?}", err);
+                                    error!("{:?}", error_msg);
+                                })
+                                .and_then(
+                                    move |amounts: Vec<BigUint>| {
+                                        let connector_amount = amounts[0].clone();
+                                        let engine_amount = amounts[1].clone();
+                                        // Scale the amount settled by the
+                                        // connector back up to our scale
+                                        let scaled_connector_amount = connector_amount
+                                            .normalize_scale(ConvertDetails {
+                                                from: quantity.scale,
+                                                to: engine_scale,
+                                            });
+
+                                        let diff: BigUint = engine_amount - scaled_connector_amount;
+                                        if diff > BigUint::from(0u32) {
```

Comment:
Import `Zero` from `num_traits`
```suggestion
                                        if diff > Zero::zero() {
```